geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+        if not isinstance(params, InterpretationSectionParams):
+            raise TypeError(
+                "Interpretation section must be an InterpretationSectionParams"
+                f"or a dict, {type(params)} provided."
+            )
+        return params
+
+
+class InterpretationSection(Group):
+    """The type for the basic Container group."""
+
+    _attribute_map = Group._attribute_map.copy()
+    _attribute_map.update(
+        {
+            "Can add group": "can_add_group",
+        }
+    )
+
+    _TYPE_UID = UUID(fields=(0x27EE59E1, 0xB1CE, 0x41EC, 0x8A, 0x86, 0x8BD3D229E198))
+    _default_name = "Interpretation Section"
+
+    def __init__(self, **kwargs):
+        self._can_add_group: bool = False
+        self._interpretation_curves: list[Curve] | None = None
+        self._interpretation_sections: list[InterpretationSectionParams] | None = None
+        self._section_object_id: Slicer | None = None
+
+        super().__init__(**kwargs)
+
+    def _load_attributes(self):
+        """
+        Load the data from the metadata only to please mypy.
+        """
+        _ = self.interpretation_curves
+        _ = self.interpretation_sections
+        _ = self.section_object_id
+
+    def _load_from_metadata(self, key: str) -> Any:
+        """
+        Load the metadata from the key.
+
+        :param key: the key to load the metadata from.
+
+        :return: The value store in the metadata if exists.
+        """
+        metadata = self.metadata if self.metadata is not None else {}
+        return metadata.get(key, None)
+
+    def _update_to_metadata(self, key: str, values: list | None):
+        """
+        Update the metadata with the interpretation curves.
+
+        :param key: The key to update.
+        :param values: The values to update.
+        """
+        if values:
+            self.update_metadata(
+                {
+                    key: [
+                        value.map_back
+                        if isinstance(value, InterpretationSectionParams)
+                        else str(value.uid)
+                        for value in values
+                    ]
+                }
+            )
+        else:
+            setattr(self, key, None)
+            self.update_metadata({key: None})
+
+    def _verify_object(self, value, key: str):
+        """
+        Get the object and verify that it is a children of the group.
+
+        :param value: The value to verify.
+        :param key: The default name of the expected object type.
+
+        :return: The verified object
+        """
+        if isinstance(value, str | UUID):
+            value = str2uuid(value)
+            value = self.workspace.get_entity(value)[0]
+
+        if getattr(value, "_default_name", None) != key:
+            raise TypeError(
+                f"The {value} object must be a {key} object."
+                f" {type(value)} provided."
+            )
+
+        return value
```

Comment:
A lot of code to just check `isinstance(obj, Curve)`. Don't really think we need this.